src/index.ts

```diff
@@ -69,6 +75,22 @@ export class MockProvider implements IMockProvider {
         }
         return Promise.resolve([this.selectedAddress])
 
+      case 'wallet_requestPermissions': {
+        if (!params[0]) return Promise.reject(new Error('Invalid method parameter(s).'))
+
+        if (!params[0].eth_accounts) return Promise.reject(new Error(`The method "${Object.keys(params[0])[0]}" does not exist / is not available.`))
+
+        const permissions: Web3WalletPermission[] = [{ parentCapability: 'eth_accounts', date: Date.now() }]
+
+        if (this.setup.manualConfirmEnable) {
+          return new Promise((resolve, reject) => {
+            this.acceptEnable = resolve
+            this.rejectEnable = reject
+          }).then(() => permissions)
```

Comment:
Can we combine this with the one from `eth_accounts` (line 71) so we don't have repeated code? 

Perhaps something like this:

```ts
private handleManualConfirm(response: any): Promise<any> {
   // create the promise and then return the response
}
```

That will make it easier to extend the confirm/reject to other methods.

Reply:
👌🏻